### PR TITLE
Convert jetty-httpclient Groovy tests to Java

### PR DESCRIPTION
**Pull request by pg-yang**
related https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/7195

### CODE REVIEW

File: instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/test/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/JettyHttpClient9LibraryTest.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+public class JettyHttpClient9LibraryTest extends AbstractJettyClient9Test {
+
+  @RegisterExtension
+  static final InstrumentationExtension extension = HttpClientInstrumentationExtension.forLibrary();
+
+  @Override
+  protected HttpClient createStandardClient() {
+    return JettyClientTelemetry.builder(testing.getOpenTelemetry())
+        .setCapturedRequestHeaders(Arrays.asList(AbstractHttpClientTest.TEST_REQUEST_HEADER))
+        .setCapturedResponseHeaders(Arrays.asList(AbstractHttpClientTest.TEST_RESPONSE_HEADER))
```
Comment (mateuszrzeszutek) [inline instrumentation/jetty-httpclient/jetty-httpclient-9.2/library/src/test/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/JettyHttpClient9LibraryTest.java:25]: Nit: you can use `Collections.singletonList()` here
```suggestion
        .setCapturedRequestHeaders(singletonList(AbstractHttpClientTest.TEST_REQUEST_HEADER))
        .setCapturedResponseHeaders(singletonList(AbstractHttpClientTest.TEST_RESPONSE_HEADER))
```

File: instrumentation/jetty-httpclient/jetty-httpclient-9.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/JettyHttpClient9AgentTest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+package io.opentelemetry.javaagent.instrumentation.jetty.httpclient.v9_2;
+
+import io.opentelemetry.instrumentation.jetty.httpclient.v9_2.AbstractJettyClient9Test;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.http.HttpClientInstrumentationExtension;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+public class JettyHttpClient9AgentTest extends AbstractJettyClient9Test {
```
Comment (mateuszrzeszutek) [inline instrumentation/jetty-httpclient/jetty-httpclient-9.2/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/JettyHttpClient9AgentTest.java:15]: JUnit 5 test classes don't need to be public
```suggestion
class JettyHttpClient9AgentTest extends AbstractJettyClient9Test {
```

File: instrumentation/jetty-httpclient/jetty-httpclient-9.2/testing/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/AbstractJettyClient9Test.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+  @AfterEach
+  public void after() throws Exception {
+    client.stop();
+    httpsClient.stop();
+  }
+
+  @Override
+  protected void configure(HttpClientTestOptions.Builder optionsBuilder) {
+    optionsBuilder.disableTestRedirects().setUserAgent("Jetty");
```
Comment (mateuszrzeszutek) [inline instrumentation/jetty-httpclient/jetty-httpclient-9.2/testing/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/AbstractJettyClient9Test.java:50]:
```suggestion
    optionsBuilder.disableTestRedirects().setUserAgent(USER_AGENT);
```

File: instrumentation/jetty-httpclient/jetty-httpclient-9.2/testing/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/AbstractJettyClient9Test.java [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+
+public abstract class AbstractJettyClient9Test extends AbstractHttpClientTest<Request> {
+
+  private HttpClient client;
+  private HttpClient httpsClient;
+  private static final String USER_AGENT = "Jetty";
```
Comment (mateuszrzeszutek) [inline instrumentation/jetty-httpclient/jetty-httpclient-9.2/testing/src/main/java/io/opentelemetry/instrumentation/jetty/httpclient/v9_2/AbstractJettyClient9Test.java:27]: Can you put static fields above class fields?
```suggestion
  private static final String USER_AGENT = "Jetty";

  private HttpClient client;
  private HttpClient httpsClient;
```

**Review by mateuszrzeszutek — COMMENTED** [review]
Thanks @pg-yang !
This is looking great, I left just a couple minor comments.

**Comment by pg-yang** [comment]
Hello, thanks for your suggestions. I think it's OK for now. And I have rebased the branch

**Review by mateuszrzeszutek — APPROVED** [review]
Thanks @pg-yang !

**Pull request merged**